R/apa_print_emm_lsm.R

```diff
@@ -68,143 +69,186 @@ apa_print.summary_emm <- function(
   )
-  contrast_table <- rename_column(contrast_table, "SE", "std.error")
 
-  contrast_table$estimate <- printnum(contrast_table$estimate, ...)
+  ## Add variable labels
+  tidy_x_labels <- c(estimate = paste0("$", est_name, "$"))
 
-  if(p_supplied) {
-    contrast_table <- rename_column(contrast_table, c("t.ratio", "z.ratio"), "statistic")
-    contrast_table$statistic <- printnum(contrast_table$statistic)
-    contrast_table$df <- printnum(contrast_table$df, digits = dfdigits)
-    contrast_table$p.value <- printp(contrast_table$p.value)
+  if("contrast" %in% factors) {
+    tidy_x_labels <- c(tidy_x_labels, contrast = "Contrast")
   }
 
   if(ci_supplied) {
-    contrast_table <- rename_column(contrast_table, c("lower.CL", "asymp.LCL"), "ll")
-    contrast_table <- rename_column(contrast_table, c("upper.CL", "asymp.UCL"), "ul")
-
-    ci_table <- data.frame(conf.int = unlist(print_confint(matrix(c(contrast_table$ll, contrast_table$ul), ncol = 2), margin = 2, conf_level = NULL, ...)))
-    contrast_table$std.error <- NULL
-    contrast_table$ll <- NULL
-    contrast_table$ul <- NULL
-
-    contrast_table <- cbind(
-      contrast_table[, 1:which(colnames(contrast_table) == "estimate")]
-      , ci_table
-      , contrast_table[, c(df_colname, stat_colnames)] # Will be NULL if not supplied
-    )
-    contrast_table$conf.int <- as.character(contrast_table$conf.int)
-    # contrast_table <- rename_column(contrast_table, "confint", "conf.int")
-  } else {
-    contrast_table$std.error <- NULL
+    tidy_x_labels <- c(tidy_x_labels, conf.int = conf_level)
   }
 
-  ## Set variable labels
   if(p_supplied) {
+    if(all(tidy_x$df == "$\\infty$")) {
+      test_stat <- "z"
+      tidy_x$df <- NULL
+    } else if(!multiple_df) {
+      test_stat <- paste0("t(", unique(tidy_x$df), ")")
+      tidy_x$df <- NULL
+    } else {
+      test_stat <- "t"
+    }
+
+    tidy_x_labels <- c(tidy_x_labels, statistic = paste0("$", test_stat, "$"))
+
+    if("null.value" %in% names(tidy_x)) {
+      tidy_x_labels <- c(tidy_x_labels, null.value = "$\\theta_0$")
+    }
+
     if(multiple_df) { # Put df in column heading
-      # colnames(contrast_table) <- c("estimate", "conf.int", "statistic", "df", "p.value")
-      # contrast_table$ci <- as.character(contrast_table$ci)
-      variable_label(contrast_table) <- c(
-        estimate = paste0("$", est_name, "$")
-        , conf.int = conf_level
-        , statistic = "$t$"
-        , df = "$\\mathit{df}$"
-        , p.value = "$p$"
-      )
+      tidy_x_labels <- c(tidy_x_labels, df = "$\\mathit{df}$")
     }
-  } else {
-    # colnames(contrast_table) <- c("estimate", "conf.int")
-    # contrast_table$ci <- as.character(contrast_table$ci)
-    variable_label(contrast_table) <- c(
-      estimate = paste0("$", est_name, "$")
-      , conf.int = conf_level
+
+    variable_labels(tidy_x[[p_value]]) <- switch(
+      p_value
+      , "p.value" = "$p$"
+      , "adj.p.value" = "$p_{adj}$"
```

Comment:
I just stumbled again over column name `adj.p.value`, which I really despise, but checked the `apa_print.glht()` and saw that we did that atrocity there, too. :D So I will just shut my mouth. Still, I think we should consider changing the label: `p_{adj}` is obviously wrong, it would (at the very least) have to be `p_\\mathit{adj}`. I would even prefer roman letters, and, if this is possible, to add the type of correction, e.g. `p_\\mathrm{BH}`. That would be really helpful, at least for in-text reporting. P.S: I would also love to use upper index and smaller font, such as `p^{\\mathrm{\\smallerBonferroni}`. Consider all of this to be a feature request

Reply:
Adding the specific correction is already on the list of things to do. I don't really understand why `adj.p.value` is a problem, but either way, we decided to adhere to the `broom`-glossar, which is where this is from.